x/stake/keeper/distribute.go

```diff
@@ -126,7 +128,13 @@ func (k Keeper) DistributeInBreathBlock(ctx sdk.Context, sideChainId string) sdk
 		events = events.AppendEvents(singleBatchEvents)
 	}
 
-	validators, height, found := k.GetHeightValidatorsByIndex(ctx, daysBackwardForValidatorSnapshot)
+	var daysBackward int
+	if sideChainId != types.MockSideChainIDForBeaconChain {
+		daysBackward = daysBackwardForValidatorSnapshot
+	} else {
+		daysBackward = daysBackwardForValidatorSnapshotBeaconChain
+	}
+	validators, height, found := k.GetHeightValidatorsByIndex(ctx, daysBackward)
```

Comment:
It is possible that the first day we enable the BEP, we are not going to find the `validators`. Ensure we check this corner case.

Reply:
There is an early return below this line if we can't find validators snapshot.
The impact is that the fees collected from the BEP activation height to the first breath block after activation will be left to the validators in snapshot saved in the first breath block.
The unit test already covered this case.

```
	validators, height, found := k.GetHeightValidatorsByIndex(ctx, daysBackward)
	if !found {
		return events
	}
```